ocw/lib/openqa.py

```diff
@@ -0,0 +1,79 @@
+from urllib.parse import urlparse
+import requests
+from requests.exceptions import RequestException
+
+# pylint: disable=import-error
+from openqa_client.client import OpenQA_Client
+from openqa_client.const import JOB_STATE_CANCELLED
+from openqa_client.exceptions import OpenQAClientError
+
+
+TAG_IGNORE = "pcw-ignore"
+
+
+def get_url(server):
+    if urlparse(server).scheme != "":
+        return server
+    for scheme in ("https", "http"):
+        try:
+            url = f"{scheme}://{server}"
+            got = requests.get(url, timeout=30)
+            got.raise_for_status()
+            return url
+        except RequestException:
+            pass
+    raise OpenQAClientError(f"Could not connect to server {server}")
+
+
+class OpenQA:
+    __servers = {}
+
+    def __new__(cls, **kwargs):
+        if kwargs["server"] not in cls.__servers:
+            server = get_url(kwargs["server"])
+            # Use same object for "https://openqa.suse.de", "openqa.suse.de" & "https://openqa.suse.de/"
```

Comment:
I am not sure why we need this complication . `get_url` function **always** returning url in same format so how do we expect to get something else except `https://openqa.suse.de` ?

Reply:
I mean that no matter what was used for init `OpenQA` object :
```
server1 = openQA("https://openqa.suse.de")
server2=openQA("openqa.suse.de")
server3=openQA("https://openqa.suse.de/")
```

`__servers` array suppose to get exactly same , than later we don't need to play a guess game here

Reply:
> I am not sure why we need this complication . `get_url` function **always** returning url in same format so how do we expect to get something else except `https://openqa.suse.de` ?

I want to avoid calling `get_url()` everytime `OpenQA` is called with the hostname without the scheme because `get_url()`  tries to guess the protocol.  So either we prepare for it here or use a `lru_cache`  decorator in `get_url`.

Reply:
ok I see , make sense than